### PR TITLE
clean up delta xds

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

File: pilot/pkg/xds/delta.go [hunk 1]
```diff
@@ -287,9 +288,7 @@ func (s *DiscoveryServer) processDeltaRequest(req *discovery.DeltaDiscoveryReque
 
 	subs := sets.New(req.ResourceNamesSubscribe...).Delete("*")
 	// InitialResourceVersions are essential subscriptions on the first request, since we don't care about the version
-	for k := range req.InitialResourceVersions {
-		subs.Insert(k)
-	}
+	subs.InsertAll(maps.Keys(req.InitialResourceVersions)...)
```
Comment (howardjohn) [inline pilot/pkg/xds/delta.go:291]: This is a performance regression, it allocates an intermediate slice we do not need. IMO not worth a 1 line cleanup

**Comment by hzxuzhonghu** [comment]
Will work on it after https://github.com/istio/istio/pull/46882

**Comment by hzxuzhonghu** [comment]
ping @howardjohn This is ready to go, Have no behavior change

**Review by howardjohn — APPROVED** [review]

**Pull request merged**